sqlmesh/core/snapshot/evaluator.py

```diff
@@ -259,14 +259,25 @@ def create(
         """
         snapshots_with_table_names = defaultdict(set)
         tables_by_schema = defaultdict(set)
+        table_deployability: t.Dict[str, bool] = {}
         for snapshot in target_snapshots:
             if not snapshot.is_model or snapshot.is_symbolic:
                 continue
-            for is_deployable in (True, False):
+            deployability_flags = (
+                [True] if (not snapshot.is_managed or not snapshot.reuses_previous_version) else []
+            )
+            if (
+                snapshot.reuses_previous_version
+                or snapshot.is_managed
+                or (deployability_index and not deployability_index.is_deployable(snapshot))
+            ):
+                deployability_flags.append(False)
+            for is_deployable in deployability_flags:
                 table = exp.to_table(
                     snapshot.table_name(is_deployable), dialect=snapshot.model.dialect
                 )
                 snapshots_with_table_names[snapshot].add(table.name)
+                table_deployability[snapshot.name + table.name] = is_deployable
```

Comment:
I think table names are unique enough so you can just use them as a key

Reply:
Oh right will change this